be/src/storage/rowset/segment_rewriter.cpp

```diff
@@ -2,18 +2,116 @@
 
 #include "segment_rewriter.h"
 
-namespace starrocks {
+#include "gutil/strings/substitute.h"
+#include "storage/fs/fs_util.h"
+#include "storage/rowset/segment_writer.h"
+#include "storage/vectorized/chunk_helper.h"
+#include "util/crc32c.h"
+#include "util/slice.h"
 
+namespace starrocks {
 namespace segment_v2 {
 
 SegmentRewriter::SegmentRewriter() {}
 
 SegmentRewriter::~SegmentRewriter() {}
 
+Status parse_segment_footer(std::unique_ptr<fs::ReadableBlock>* rblock, SegmentFooterPB* footer,
+                            size_t* footer_length) {
+    std::string buff;
+    raw::stl_string_resize_uninitialized(&buff, 12);
+
+    uint64_t file_size;
+    RETURN_IF_ERROR((*rblock)->size(&file_size));
+    RETURN_IF_ERROR((*rblock)->read(file_size - buff.size(), buff));
+
+    *footer_length = UNALIGNED_LOAD32(buff.data() + buff.size() - 12);
+    const uint32_t checksum = UNALIGNED_LOAD32(buff.data() + buff.size() - 8);
+    const uint32_t magic_number = UNALIGNED_LOAD32(buff.data() + buff.size() - 4);
+
+    if (magic_number != UNALIGNED_LOAD32(k_segment_magic)) {
+        return Status::Corruption(
+                strings::Substitute("Bad segment file $0: magic number not match", (*rblock)->path()));
+    }
+
+    if (file_size < 12 + *footer_length) {
+        return Status::Corruption(strings::Substitute("Bad segment file $0: file size $1 < $2", (*rblock)->path(),
+                                                      file_size, 12 + *footer_length));
+    }
+
+    raw::stl_string_resize_uninitialized(&buff, *footer_length);
+    RETURN_IF_ERROR((*rblock)->read(file_size - buff.size() - 12, buff));
+    uint32_t actual_checksum = crc32c::Value(buff.data(), buff.size());
+    if (actual_checksum != checksum) {
+        return Status::Corruption(
+                strings::Substitute("Bad segment file $0: footer checksum not match, actual=$1 vs expect=$2",
+                                    (*rblock)->path(), actual_checksum, checksum));
+    }
+
+    if (!footer->ParseFromArray(buff.data(), buff.size())) {
+        return Status::Corruption(
+                strings::Substitute("Bad segment file $0: failed to parse footer", (*rblock)->path()));
+    }
+
+    return Status::OK();
+}
```

Comment:
similar with https://github.com/StarRocks/starrocks/blob/d78926e4ad897c016cbde18a4a2c394d4d5b84fb/be/src/storage/rowset/segment.cpp#L129-L206, can reuse?

Reply:
> similar with
> 
> https://github.com/StarRocks/starrocks/blob/d78926e4ad897c016cbde18a4a2c394d4d5b84fb/be/src/storage/rowset/segment.cpp#L129-L206
> 
> , can reuse?

done